### PR TITLE
Merge KernelConfig with Kernel

### PR DESCRIPTION
**Pull request by mkarle**
### Motivation and Context

KernelConfig isn't needed as it's less pythonic and is not used independently from Kernel.  This moves all its methods and members into Kernel.

### Description

- Removed KernelConfig
- Put KernelConfig's methods and variables into Kernel
- Ran unit and end-to-end tests
- Did not run notebooks because they are out of date with the current repo.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by awharrison-28 — APPROVED** [review]

**Comment by awharrison-28** [comment]
Looks like there are conflicts to resolve in notebook 5, otherwise, LGTM

File: python/semantic_kernel/kernel.py [hunk 1]
```diff
@@ -248,6 +265,237 @@ def import_skill(
...
+        self._text_embedding_generation_services[service_id] = (
+            service if isinstance(service, Callable) else lambda _: service
+        )
+        if self._default_text_embedding_generation_service is None:
+            self._default_text_embedding_generation_service = service_id
+
+        return self
+
+    # TODO: look harder at retry stuff
+
```
Comment (shawncal) [inline python/semantic_kernel/kernel.py:375]:
```suggestion
```

Comment (shawncal) [inline python/semantic_kernel/kernel.py:375]:
```suggestion
```

**Review by shawncal — APPROVED** [review]

**Review by shawncal — DISMISSED** [review]

File: python/semantic_kernel/kernel.py [hunk 2]
```diff
@@ -41,10 +47,11 @@
 from semantic_kernel.utils.null_logger import NullLogger
 from semantic_kernel.utils.validation import validate_function_name, validate_skill_name
 
+T = TypeVar("T")
+
```
Comment (shawncal) [inline python/semantic_kernel/kernel.py:51]:
```suggestion
T = TypeVar("T")
```

File: python/semantic_kernel/kernel.py [hunk 3]
```diff
@@ -41,10 +47,11 @@
...
 
 class Kernel(KernelBase, KernelExtensions):
```
Comment (shawncal) [inline python/semantic_kernel/kernel.py:53]:
```suggestion
from semantic_kernel.utils.validation import validate_function_name, validate_skill_name

T = TypeVar("T")

class Kernel(KernelBase, KernelExtensions):
```

**Review by shawncal — APPROVED** [review]

**Comment by nsteblay** [comment]
I'm far from a strong Python engineer but have been cloning this repo to learn about the Semantic Kernel.  This change breaks all the Python notebooks.  To continue my learning I need to reverse the changes that were made.  I guess the notebooks aren't as "Pythonic" as they should be, whatever that means, but at least they worked.  Given that this is code for introducing some of the basics I would highly recommend that the code work.  This change provides no value at all, exactly the opposite.

**Pull request merged**